clippy_lints/src/lib.rs

```diff
@@ -964,7 +964,23 @@ pub fn register_plugins(store: &mut rustc_lint::LintStore, sess: &Session, conf:
     store.register_late_pass(|| box strings::StringAdd);
     store.register_late_pass(|| box implicit_return::ImplicitReturn);
     store.register_late_pass(|| box implicit_saturating_sub::ImplicitSaturatingSub);
-    store.register_late_pass(|| box methods::Methods);
+
+    let parsed_msrv = conf.msrv.as_ref().and_then(|s| {
+        parse_msrv(s, None, None).or_else(|| {
+            sess.err(&format!("`{}` is not a valid Rust version", s));
+            None
+        })
+    });
+
```

Comment:
@flip1995 I was able to remove the `as_ref()` calls here but still had to `clone()` the msrv multiple times as there were errors about moving an outer variable to a closure. Can this be avoided?

` cannot move out of `msrv`, a captured variable in an `Fn` closure1`

Reply:
No, it's expected that the Version has to be cloned. So that's fine.